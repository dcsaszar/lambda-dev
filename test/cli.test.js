const CliTest = require('command-line-test');

describe('cli', () => {
  test('Without arguments', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });

  test('Build without entry argument', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js build');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });

  test('Build without target argument', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js build test/fixtures');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });

  test('Serve without entry argument', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js serve');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });
});
