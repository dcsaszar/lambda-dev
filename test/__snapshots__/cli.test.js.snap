// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli Build without entry argument 1`] = `
[Error: Command failed: node cli.js build
cli.js build <entry> <target>

Bundle Lambda functions

Positionals:
  entry   Relative path to entry file or directory of functions
                                                             [string] [required]
  target  Relative path to target directory for bundled functions
                                                             [string] [required]

Options:
  --help                Show help                                      [boolean]
  --version             Show version number                            [boolean]
  --node, -n            Target Node.js version, used with @babel/preset-env
                                                      [string] [default: "8.10"]
  --webpack-config, -w  Relative path to optional Webpack configuration. The
                        default export should be an object that is merged with
                        the default webpack config, or a function that receives
                        the default configuration and returns a valid webpack
                        configuration.                                  [string]
  --watch               Control the webpack watch mode. Defaults to true for
                        serve and false for build.                     [boolean]

Not enough non-option arguments: got 0, need at least 2
]
`;

exports[`cli Build without target argument 1`] = `
[Error: Command failed: node cli.js build test/fixtures
cli.js build <entry> <target>

Bundle Lambda functions

Positionals:
  entry   Relative path to entry file or directory of functions
                                                             [string] [required]
  target  Relative path to target directory for bundled functions
                                                             [string] [required]

Options:
  --help                Show help                                      [boolean]
  --version             Show version number                            [boolean]
  --node, -n            Target Node.js version, used with @babel/preset-env
                                                      [string] [default: "8.10"]
  --webpack-config, -w  Relative path to optional Webpack configuration. The
                        default export should be an object that is merged with
                        the default webpack config, or a function that receives
                        the default configuration and returns a valid webpack
                        configuration.                                  [string]
  --watch               Control the webpack watch mode. Defaults to true for
                        serve and false for build.                     [boolean]

Not enough non-option arguments: got 1, need at least 2
]
`;

exports[`cli Serve without entry argument 1`] = `
[Error: Command failed: node cli.js serve
cli.js serve <entry>

Lambda Development Server

Positionals:
  entry  Entry file or directory of Lambda functions                  [required]

Options:
  --help                Show help                                      [boolean]
  --version             Show version number                            [boolean]
  --node, -n            Target Node.js version, used with @babel/preset-env
                                                      [string] [default: "8.10"]
  --webpack-config, -w  Relative path to optional Webpack configuration. The
                        default export should be an object that is merged with
                        the default webpack config, or a function that receives
                        the default configuration and returns a valid webpack
                        configuration.                                  [string]
  --watch               Control the webpack watch mode. Defaults to true for
                        serve and false for build.                     [boolean]
  --basePath, -b        Base pathname prepended to function filenames
                                                         [string] [default: "/"]
  --port, -p            Port for development server     [number] [default: 9000]

Not enough non-option arguments: got 0, need at least 1
]
`;

exports[`cli Without arguments 1`] = `
[Error: Command failed: node cli.js
cli.js <command>

Commands:
  cli.js serve <entry>           Lambda Development Server
  cli.js build <entry> <target>  Bundle Lambda functions

Options:
  --help                Show help                                      [boolean]
  --version             Show version number                            [boolean]
  --node, -n            Target Node.js version, used with @babel/preset-env
                                                      [string] [default: "8.10"]
  --webpack-config, -w  Relative path to optional Webpack configuration. The
                        default export should be an object that is merged with
                        the default webpack config, or a function that receives
                        the default configuration and returns a valid webpack
                        configuration.                                  [string]
  --watch               Control the webpack watch mode. Defaults to true for
                        serve and false for build.                     [boolean]

Not enough non-option arguments: got 0, need at least 1
]
`;
